app/src/main/java/com/example/kotlinmigration/MainActivity.kt

```diff
@@ -60,9 +56,11 @@ class MainActivity : AppCompatActivity() {
                     is MainViewModel.RetrofitEvent.Successful -> {
                         if(it.response != null)
                         {
-                            myProgress?.visibility = View.INVISIBLE
-                            recyclerView?.adapter = MyAdapter(it.response)
+                            binding.progress.visibility = View.INVISIBLE
+                            binding.recyclerView.adapter = MyAdapter(it.response)
+
```

Comment:
if you implement `MainViewModel.insertDataToDatabase(data: List<PostJsonItem>)` in the way I mentioned it below

you can call it here with `it.response` inside

Reply:
Implemented